plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/job/OpenShiftJobs.java

```diff
@@ -56,11 +71,39 @@ protected IStatus doRun(IProgressMonitor monitor) {
 							ConnectionProperties.PROPERTY_PROJECTS, 
 							oldProjects, 
 							newProjects);
+				}else {
+					notifyCallerOfFailure(callIfNotOK);
 				}
 				return status;
 			}
+
+			private void waitForServerToReconcileProjectDelete(Connection conn, IProject project) {
+				OpenShiftException e = null;
+				long sleep = 0;
+				do {
+					try {
+						conn.getResource(project);
+						Thread.sleep(PROJECT_DELETE_DELAY);
+						sleep = sleep + PROJECT_DELETE_DELAY;
+					} catch (InterruptedException e1) {
+					} catch(OpenShiftException ex) {
+						e = ex;
```

Comment:
shouldn't we make sure that the exception we get really is an exception caused by the fact that the project isnt there any longer?